ast/case_statement.go

```diff
@@ -0,0 +1,53 @@
+package ast
+
+import (
+	"bytes"
+)
+
+type CaseStatement struct {
+	*Meta
+	Test       *InfixExpression
+	Statements []Statement
+}
+
+func (s *CaseStatement) statement()     {}
+func (s *CaseStatement) GetMeta() *Meta { return s.Meta }
+func (s *CaseStatement) FinalStatement() Statement {
+	return s.Statements[len(s.Statements)-1]
+}
+func (s *CaseStatement) IsBreak() bool {
+	_, ok := s.FinalStatement().(*BreakStatement)
+	return ok
+}
+func (s *CaseStatement) IsFallthrough() bool {
+	_, ok := s.FinalStatement().(*FallthroughStatement)
+	return ok
+}
+func (s *CaseStatement) TestEqual(other *CaseStatement) bool {
+	if s.Test == nil || other.Test == nil || s.Test.Operator != other.Test.Operator {
+		return false
+	}
+	return s.Test.Right.String() == other.Test.Right.String()
+}
```

Comment:
`AST` must have only syntax tree and token representation so we should not have any logic here.
An `interpreter` package should implement this logic

Reply:
Ok makes sense. I was looking at the AST type as just a convenient place to put some helper methods that would be used by the parser when validating the cases. Is your view that these kinds of checks should be done in the interpreter (and by extension in the linter) rather than in the parser?

Specifically checking for duplicate cases and ensuring the final case ends with a `break`.

Reply:
Removed the helper methods from the CaseStatement type. To start I left the syntax validation in the parser if you'd prefer for that validation to be moved to the interpreter I can do that as well.